decoder.go

```diff
@@ -208,15 +214,13 @@ func (d *decoder) decode(nalu []byte) (*avFrameWrapper, error) {
 	avPacket.size = C.int(len(nalu))
 	res := C.avcodec_send_packet(d.codecCtx, &avPacket)
 	if res < 0 {
-		//nolint:nilnil // TODO RSDK-8575: change to not nil, nil
-		return nil, nil
+		return nil, newRecoverableAVError(int(res), "error sending packet to the decoder")
 	}
 
 	// receive frame if available
 	res = C.avcodec_receive_frame(d.codecCtx, d.src)
 	if res < 0 {
-		//nolint:nilnil // TODO RSDK-8575: change to not nil, nil
-		return nil, nil
+		return nil, newRecoverableAVError(int(res), "error receiving decoded frame from the decoder")
```

Comment:
Do we want to lookup FFmpeg error message (av_strerror) associated with res int?

Reply:
Took a stab at implementing including av error strings everywhere they are provide-able

Reply:
Nice, LGTM